Mw solver example V1 (#2011)
chaos-polymtl/lethe: doc/source/examples/multiphysics/EM-waves/EM-waves.rst

Reviewer: You can rename the folder and the file to correspond to the example title, here "waveguide"

> =============================

Reviewer: You can add the license at beginning of the file
```suggestion
..
  SPDX-FileCopyrightText: Copyright (c) 2026 The Lethe Authors
  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception OR LGPL-2.1-or-later

=============================
```

Waveguide

> =============================

Reviewer 2: Generally you will want as many "=" character as the length of the "Waveguide" word above. Otherwise you will get a documentation warning

> This example showcases an electromagnetic problem that we can solve analytically, allowing us to test the results of our simulation. It is the first example using the DPG method (Discontinuous Petrov-Gallerkin method) of the time-harmonic section of lethe.

Reviewer: I would move here some of the description you put in your parameter file.
```suggestion
This example showcases the propagation of an electromagnetic excitation in a waveguide with perfectly conducting walls by solving the :doc:`Time Harmonic Maxwell equations<../../../theory/multiphysics/electromagnetism/time_harmonic>`.
```

> This example showcases an electromagnetic problem that we can solve analytically, allowing us to test the results of our simulation. It is the first example using the DPG method (Discontinuous Petrov-Gallerkin method) of the time-harmonic section of lethe.

Reviewer 3: ```suggestion
This example is used to verify the implementation of the electromagnetic solver. It considers the propagation of an electromagnetic wave in a waveguide, a problem for which an analytical solution is available, thereby allowing us to assess whether the method recovers the expected order of convergence.
```

-----------------------------
Features
-----------------------------

- Solvers: ``lethe-fluid``
- Steady-state problem

> - Time-harmonic functions

Reviewer: ```suggestion
- Solvers: ``lethe-fluid`` or ``lethe-fluid-matrix-free``
- Steady-state problem
- Use of the discontinous Petrov-Galerkin (DPG) time-harmonic Maxwell solver
```

----------------------------
Files Used In This Example
----------------------------

All files mentioned below are located in the example's folder (``examples/multiphysics/waveguide``)

> - Base case parameter file (:math:`\mathrm{f=2.45 GHz}`): ``parameters.prm``

Reviewer: with the previous comment
```suggestion
- Base case parameter file (:math:`f=\mathrm{2.45 GHz}`): ``waveguide.prm``
```

> - Postprocessing Python script for the :math:`\mathrm{Re}=400` case: ``convergence.py``

Reviewer 3: I don't see why it talks about the Reynold number, probably a typo/copy-past. You can remove this if you don't create a post-processing file.

---------------------------
Description of the Case
---------------------------

Geometry
~~~~~~~~

> The waveguide at stake has a square cross section.It consists of four perfectly reflective walls and two empty sections: one for the wave’s entry and the other for its exit. The wave propagates along the z-axis. It is described on the next figure:

Reviewer: ```suggestion
The simulated waveguide has a square cross section. It is made of four perfectly reflective lateral walls, an inlet port for the wave’s entry, and an outlet for its exit  as shown in the figure below. The wave propagates along the :math:`z`-axis.
```

.. image:: images/schematic.png
    :alt: schematic
    :align: center
    :name: geometry
    :width: 500

> The dimensions chosen here are a width a = 0.25m,  a height b = 0.25m and a length L = 1m.

Reviewer: The length $L$ is not shown in the figure, can you add it please?

> The dimensions chosen here are a width a = 0.25m,  a height b = 0.25m and a length L = 1m.

Reviewer: ```suggestion
The dimensions chosen here are a width :math:`a = 0.25 \, \mathrm{m}`,  a height :math:`b = 0.25\, \mathrm{m}` and a length :math:`L = 1\, \mathrm{m}`.
```

> Physical problem

Reviewer: ```suggestion
Physical Problem
```

~~~~~~~~~~~~~~~~

> This simulation calculates the stationnary electromagnetic field in the waveguide. Even though the waveguide is empty, we aim to calculate the fields in matter when we will add the particle flow in our problem. Thus, we start with Maxwell's equations in matter, where :math:`E` and :math:`H` denote the electric and magnetic fields:

Reviewer: Since E and H are vector fields
```suggestion
This simulation computes the stationary electromagnetic field in the waveguide. Even though the waveguide is empty, we aim to calculate the fields in matter when we will add the particle flow in our problem. Thus, we start with Maxwell's equations in matter, where :math:`\mathbf{E}` and :math:`\mathbf{H}` denote the electric and magnetic fields:
```

> This simulation calculates the stationnary electromagnetic field in the waveguide. Even though the waveguide is empty, we aim to calculate the fields in matter when we will add the particle flow in our problem. Thus, we start with Maxwell's equations in matter, where :math:`E` and :math:`H` denote the electric and magnetic fields:

Reviewer 3: In this section i would not talk about how we recover the time-harmonic equation, it would be part of the theory section. Just say something like: "This simulation computes the stationary electromagnetic field in the above waveguide with perfectly conducting metal wall and filled with void. Recall that the time-harmonic maxwell equation in their dimensionless form used in this test case are [...], where the following variable [...] represents [...].

:math:`\nabla \times \mathbf(E) = \frac{\partial (\mu \mathbf(H))}{\partial t} \quad \mathrm{(Farady's law)}`

> :math:`\nabla \times \mathbf(H) = \mathbf(J) - \frac{\partial (\varepsilon \mathbf(E))}{\partial t} \quad \mathrm{(Ampère-Maxwell's law)}`

Reviewer: ```suggestion
.. math::
  \begin{align*}
   \tag{Farady's law} \nabla \times \mathbf{E} &= \frac{\partial (\mu \mathbf{H})}{\partial t}\\
   \tag{Ampère-Maxwell's law} \nabla \times \mathbf{H} &= \mathbf{J} - \frac{\partial (\varepsilon \mathbf{E})}{\partial t}
  \end{align*}

with :math:`\mu` the permeability, :math:`\mathbf{J}` the current density, and :math:`\varepsilon` the permittivity.
```

Then, we use the time-harmonic ansatz : :math:`\begin{align}
\mathbf{E}(\mathbf{x}, t) &= \Re\left\{\mathbf{E}_{\text{spatial}}(\mathbf{x})\, e^{-i\omega t}\right\} \\
\mathbf{H}(\mathbf{x}, t) &= \Re\left\{\mathbf{H}_{\text{spatial}}(\mathbf{x})\, e^{-i\omega t}\right\}

> \end{align}`

Reviewer: ```suggestion
Then, we use the time-harmonic ansatz:

.. math::
  \begin{align*}
  \mathbf{E}(\mathbf{x}, t) &= \Re\left\{\mathbf{E}_{\text{spatial}}(\mathbf{x})\, e^{-i\omega t}\right\} \\
  \mathbf{H}(\mathbf{x}, t) &= \Re\left\{\mathbf{H}_{\text{spatial}}(\mathbf{x})\, e^{-i\omega t}\right\}
  \end{align*}
```

> This form of solution introduces :math:`\omega`, the angular frequency of the electromagnetic wave, and 12 unknowns. In fact, for each of the :math:`E` and :math:`H` fields, we calculate the real and imaginary parts in each of the three spatial directions. 

Reviewer: ```suggestion
This form of solution introduces :math:`\omega`, the angular frequency of the electromagnetic wave, and 12 unknowns. In fact, for each of the :math:`\mathbf{E}` and :math:`\mathbf{H}` fields, we compute the real and imaginary parts in each of the three spatial directions.
```

> Boundary conditions

Reviewer: ```suggestion
Boundary Conditions
```

~~~~~~~~~~~~~~~~~~~

> .. warning::

Reviewer: ```suggestion
.. note::
```

>     From now on, the equations are written in a dimensionless form.

Reviewer 3: ```suggestion
    Lethe time-harmonic solver always solve the set of equation in a dimensionless form, so that is the convention that will be used in what follows. 
```

> There are three types of boundary conditions in this problem, this explains why the surfaces of the waveguide are sorted in three groups. There is first the inlet :math:`\Gamma_1`, the outlet :math:`\Gamma_2` and finally the metal walls :math:`\Gamma_3`.

Reviewer 3: Add the boundary condition of the waveguide.prm in this section and make links when you feel necessary.  For example you can describe how the waveguide inlet boundary condition is used and the relevant parameters.

- Inlet :math:`\Gamma_1`: waveguide port boundary condition at the inlet of the waveguide to excite the :math:`\mathrm{TE}_{mn}` mode at z = 0

> - Outlet :math:`\Gamma_2`: impedance matching boundary condition to minimize the reflections (the waveguide is theorically infinite). We calculated the dimensionless admittance in such conditions: :math:`\frac{1}{Z_s}=\frac{k_zc}{\omega \mu_r}`

Reviewer: Try to introduce the variables when they appear for the first time :)

```suggestion
- Inlet :math:`\Gamma_1`: waveguide port boundary condition at the inlet of the waveguide to excite the :math:`\mathrm{TE}_{mn}` mode at :math:`z` = 0
- Outlet :math:`\Gamma_2`: impedance matching boundary condition to minimize the reflections (the waveguide is theoretically infinite). The dimensionless admittance in such conditions is:

  .. math::
    \frac{1}{Z_s}=\frac{k_\mathrm{z}c}{\omega \mu_\mathrm{r}}

  where :math:`k_\mathrm{z}` is the component of the wavenumber in the direction of wave propagation (:math:`z`), :math:`c` is the speed of light, and :math:`\mu_\mathrm{r}` is the relative permeability of the matter.
```

- Metal walls :math:`\Gamma_3`: perfect electric conductor (PEC) boundary conditions

This results in the three following equations:

.. math::
    \begin{align}
    \nabla \times \mathbf{H} + \frac{k_z}{\omega \mu_r} \mathbf{n} \times (\mathbf{E} \times \mathbf{n}) &=  \nabla \times \mathbf{H}_{\mathrm{TE}_{mn}} + \frac{k_z}{\omega \mu_r} \mathbf{n} \times (\mathbf{E}_{\mathrm{TE}_{mn}} \times \mathbf{n}) \quad (\text{on } \Gamma_1) \\
    \nabla \times \mathbf{H} + \frac{k_z}{\omega \mu_r} \mathbf{n} \times (\mathbf{E} \times \mathbf{n}) = 0 \quad (\text{on } \Gamma_2) \\
    \mathbf{n} \times \mathbf{E} = 0 \quad (\text{on } \Gamma_3)

>     \end{align}

Reviewer: ```suggestion
These translate into the three following equations:

.. math::
    \begin{align*}
    \tag{on $\Gamma_1$} \nabla \times \mathbf{H} + \frac{k_\mathrm{z}}{\omega \mu_\mathrm{r}} \mathbf{n} \times (\mathbf{E} \times \mathbf{n}) &=  \nabla \times \mathbf{H}_{\mathrm{TE}_{mn}} + \frac{k_\mathrm{z}}{\omega \mu_\mathrm{r}} \mathbf{n} \times (\mathbf{E}_{\mathrm{TE}_{mn}} \times \mathbf{n})\\
    \tag{on $\Gamma_2$}\nabla \times \mathbf{H} + \frac{k_\mathrm{z}}{\omega \mu_\mathrm{r}} \mathbf{n} \times (\mathbf{E} \times \mathbf{n}) &= 0 \\
    \tag{on $\Gamma_3$}\mathbf{n} \times \mathbf{E} &= 0
    \end{align*}
```

.. note:: 

>     :math:`\mathrm{TE_mn}` refers to "Tranverse electric mode". This means that regardless of the values of m and n, the z-component of the vector E is always zero. Therefore, the pair (m,n) refers to the fact that the x-component of the electric field is excited in the m-th mode and the y-component of the electric field is excited on the n-th mode. 

Reviewer: ```suggestion
    :math:`\mathrm{TE}_{mn}` refers to "Transverse electric mode". This means that regardless of the values of :math:`m` and :math:`n`, the :math:`z`-component of :math:`\mathbf{E}` is always zero. Therefore, the pair :math:`(m,n)` refers to the :math:`x`-component of the electric field being excited in the :math:`m`-th mode and the :math:`y`-component of the electric field being excited on the :math:`n`-th mode. 
```

Dimensionless analytical solution
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Solution of a :math:`\mathrm{TE_mn}` mode that we use for this test case is given by:

Reviewer: ```suggestion
Dimensionless Analytical Solution
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Solution of a :math:`\mathrm{TE}_{mn}` mode that we use for this test case is given by:
```

> Solution of a :math:`\mathrm{TE_mn}` mode that we use for this test case is given by:

Reviewer 3: ```suggestion
The solution for a :math:`\mathrm{TE_mn}` mode, which will be used to assess convergence for this test case, is given by:
```

.. math::
    \mathbf{E} = i\frac{\omega\mu_r}{k_c^2} \begin{bmatrix}
    -k_y \cos(k_x x) \sin(k_y y) \\
    k_x \sin(k_x x) \cos(k_y y) \\
    0
    \end{bmatrix} e^{ik_z z}

.. math::
    \mathbf{H} =  \begin{bmatrix}
    -i\frac{k_z k_x}{k_c^2} \sin(k_x x) \cos(k_y y) \\
    -i\frac{k_z k_y}{k_c^2} \cos(k_x x) \sin(k_y y) \\
    \cos(k_x x)\cos(k_y y)

>     \end{bmatrix} e^{ik_z z}

Reviewer: ```suggestion
.. math::
    \mathbf{E} = i\frac{\omega\mu_r}{k_c^2} \begin{bmatrix}
    -k_\mathrm{y} \cos(k_\mathrm{x} x) \sin(k_\mathrm{y} y) \\
    k_\mathrm{x} \sin(k_\mathrm{x} x) \cos(k_\mathrm{y} y) \\
    0
    \end{bmatrix} e^{ik_\mathrm{z} z}

.. math::
    \mathbf{H} =  \begin{bmatrix}
    -i\frac{k_\mathrm{z} k_\mathrm{x}}{k_c^2} \sin(k_\mathrm{x} x) \cos(k_\mathrm{y} y) \\
    -i\frac{k_\mathrm{z} k_\mathrm{y}}{k_c^2} \cos(k_\mathrm{x} x) \sin(k_\mathrm{y} y) \\
    \cos(k_\mathrm{x} x)\cos(k_\mathrm{y} y)
    \end{bmatrix} e^{ik_\mathrm{z} z}
```

    
With the dimensionless parameters:

> :math:`k_x = \frac{Lm\pi}{a} \ , \ k_y = \frac{Ln\pi}{b} \ , \ k_c^2 = L^2(k_x^2 + k_y^2) \ \text{and} \ k_z = \sqrt{\omega^2 \varepsilon_{eff,r}\mu_r - k_c^2} \ , \ \omega = \frac{L2\pi f}{c}`

Reviewer: Can you introduce $k_\mathrm{x}$, $k_\mathrm{y}$, and $k_\mathrm{c}$ and maybe explain the cutoff wavenumber since this example is a but of an intro to THM.
```suggestion
:math:`k_\mathrm{x} = \frac{Lm\pi}{a} \ , \ k_\mathrm{y} = \frac{Ln\pi}{b} \ , \ k_\mathrm{c}^2 = L^2(k_\mathrm{x}^2 + k_\mathrm{y}^2) \ \text{and} \ k_\mathrm{z} = \sqrt{\omega^2 \varepsilon_\mathrm{eff,r}\mu_\mathrm{r} - k_\mathrm{c}^2} \ , \ \omega = \frac{L2\pi f}{c}`
```

> :math:`k_x = \frac{Lm\pi}{a} \ , \ k_y = \frac{Ln\pi}{b} \ , \ k_c^2 = L^2(k_x^2 + k_y^2) \ \text{and} \ k_z = \sqrt{\omega^2 \varepsilon_{eff,r}\mu_r - k_c^2} \ , \ \omega = \frac{L2\pi f}{c}`

Reviewer 3: I think you should remove the L in all the equation there and just add a ::warning:: that everything has been made dimensionless assuming Lethe use the default (aka meter).

--------------
Parameter File
--------------

> Mesh adaptation

Reviewer: ```suggestion
Mesh Adaptation
```

~~~~~~~~~~~~~~~

> The first parameter we can change is the the number n of mesh adaptations:

Reviewer: ```suggestion
The first parameter we can change is the the number ``n`` of mesh adaptations in the subsection ``simulation control``:
```

.. code-block:: text

    subsection simulation control
        set method            = steady
        set output frequency  = 1
        set output path = ./degree_1/
        set number mesh adapt = n

>     end

Reviewer: This needs to be indented with prmindent. I can show you how, if you are not familiar with it yet :)

> 

Reviewer: ```suggestion

We specify, in the subsection ``mesh adaptation``, the ``type``` to be ``uniform``:

.. code-block:: text

    subsection mesh adaptation
      set type = uniform
    end

```

.. Note:: 

>     A normal lab PC (12 cores, 64 Go RAM) can only deal with 3 mesh adaptations. 

Reviewer: ```suggestion
.. Note:: 
    A typical lab PC (12 cores, 64 Go RAM) can compute up to 3 mesh adaptations. 
```

> The following figure represents a visualation of the results on paraview with the default settings and 4 mesh adptations.

Reviewer: ```suggestion
The following figure shows the :math:`\Re(\mathbf{H}_\mathrm{z})` field of 4 increasing mesh adaptations. As expected, with decreasing element size we see better results.
```

> The following figure represents a visualation of the results on paraview with the default settings and 4 mesh adptations.

Reviewer: I am not sure if this shouldn't be in the ``Results and Discussion`` section.

Reviewer 3: Yes, i would also put it in the result section.

.. image:: images/Mesh_adaptation.png
    :alt: standing-wave-mesh
    :align: center
    :name: standing-wave-mesh
    :width: 500

> Frequency of the electromagnetic wave

Reviewer: ```suggestion
Frequency of the Electromagnetic Wave
```

> Frequency of the electromagnetic wave

Reviewer: Can you put the block of parameters of the ``time harmonic maxwell`` subsection one level higher than this one? You can also explain how the inlet is defined in the parameter file. The idea of examples is also for users to gain familiarity with the different subsections and use it for their own purposes.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> The frequency of the inlet is originally set up to :math:`f = \mathrm{2.45 GHz}` because it the nominal frequency of the MW in the industry. This value can still be modified at line 84 : ``set electromagnetic frequency  = 2.45e9``

Reviewer: ```suggestion
The frequency of the inlet is originally set to :math:`f = \mathrm{2.45}\, \mathrm{GHz}` because it the nominal frequency of microwaves (MW) in the industry. This value can still be modified at line 84 : ``set electromagnetic frequency  = 2.45e9``
```

.. Note:: 

>     Not only has the frequency to be changed, but also :math:`k_z = \sqrt{\omega^2 \varepsilon_{eff,r}\mu_r - k_x^2 - k_y^2 },\ and \ k = \sqrt{k_x^2 +k_y^2 + k_z^2}` at line 153, and the dimensionless admittance real part :math:`\Re (G)` at line 147 :math:`\Re (G) =\frac{1}{Z_s}=\frac{k_zc}{\omega \mu_r}` because they depend on :math:`f` through :math:`\omega = \frac{2\pi Lf}{c}`. 

Reviewer: ```suggestion
.. caution::
    If the frequency is changed, :math:`k_\mathrm{z} = \sqrt{\omega^2 \varepsilon_\mathrm{eff,r}\mu_\mathrm{r} - k_\mathrm{x}^2 - k_\mathrm{y}^2}` from :math:`k = \sqrt{k_\mathrm{x}^2 +k_\mathrm{y}^2 + k_\mathrm{z}^2}` at line 153, and the dimensionless admittance real part :math:`\Re (G) =\frac{1}{Z_s}=\frac{k_\mathrm{z}c}{\omega \mu_\mathrm{r}}` at line 147 need to be changed since they depend on :math:`f` through :math:`\omega = \frac{2\pi Lf}{c}`.
```

>     Not only has the frequency to be changed, but also :math:`k_z = \sqrt{\omega^2 \varepsilon_{eff,r}\mu_r - k_x^2 - k_y^2 },\ and \ k = \sqrt{k_x^2 +k_y^2 + k_z^2}` at line 153, and the dimensionless admittance real part :math:`\Re (G)` at line 147 :math:`\Re (G) =\frac{1}{Z_s}=\frac{k_zc}{\omega \mu_r}` because they depend on :math:`f` through :math:`\omega = \frac{2\pi Lf}{c}`. 

Reviewer: line 153 is an empty line

.. warning::

>     The simulation may not run for low frequencies. Indeed, if :math:`k_z^2 = \omega^2 \varepsilon_{eff,r}\mu_r - k_x^2 - k_y^2` becomes negative, the mathematical solution still exists as evanescent waves, but it does not represent a physical system.

Reviewer: ```suggestion
.. warning::
    The simulation may not run for low frequencies. Indeed, if :math:`k_\mathrm{z}^2 = \omega^2 \varepsilon_\mathrm{eff,r}\mu_\mathrm{r} - k_\mathrm{x}^2 - k_\mathrm{y}^2` becomes negative, the mathematical solution still exists as evanescent waves, but it does not represent a physical system.
```

> The different Transverse Electric mode

Reviewer 3: This is a nice section! I would merge it to the boundary conditions since it is linked to the definition of the waveguide inlet boundary.

> ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Reviewer: ```suggestion
The Different Transverse Electric Modes
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

> The :math:`\mathrm{TE_mn}` mode can be modified at lines 97 and 98, the defaultsettings are:

Reviewer: ```suggestion
The :math:`\mathrm{TE}_{mn}` mode can be modified at lines 97 and 98, the default settings are:
```

.. code-block:: text

    subsection waveguide mode
      set mode type    = TE
      set mode order m = 1
      set mode order n = 0
    end

.. Note:: 

>     Not only have the modes to be changed, but also :math:`k_z = \sqrt{(L \omega)^2 \epsilon_{eff}\mu - k_x^2 - k_y^2 },\ and \ k = \sqrt{k_x^2 +k_y^2 + k_z^2}` at line 153, and the dimensionless admittance real part :math:`\Re (G)` at line 147 :math:`\Re (G) =\frac{1}{Z_s}=\frac{k_z}{\omega \mu_r}` because they depend on :math:`(m,n)` through :math:`k_x = \frac{m \pi}{a} \ , \ k_y = \frac{n \pi}{b}`. 

Reviewer: ```suggestion
.. caution::
    If :math:`\mathrm{TE}_{mn}` is changed, :math:`k_\mathrm{z} = \sqrt{(L \omega)^2 \varepsilon_{eff}\mu - k_\mathrm{x}^2 - k_\mathrm{y}^2 }` from :math:`k = \sqrt{k_\mathrm{x}^2 +k_\mathrm{y}^2 + k_\mathrm{z}^2}` at line 153, and the dimensionless admittance real part :math:`\Re (G)=\frac{1}{Z_s}=\frac{k_\mathrm{z} c}{\omega \mu_\mathrm{r}}` at line 147 must be changed because they depend on :math:`(m,n)` through :math:`k_\mathrm{x} = \frac{m \pi}{a}` and :math:`k_\mathrm{y} = \frac{n \pi}{b}`.
```

.. warning:: 
    If you want to try high modes, you have to adapt the frequency in order for :math:`k_z` to keep a real value. Indeed, the number :math:`k_z^2 = (L \omega)^2 \epsilon_{eff}\mu - k_x^2 - k_y^2` can be negative.

> Here are paraview visuals with different modes. In order to maximise the resolution, we calculated the parameters with :math:`\lambda = L = 1m`, which is equivalent to fix the dimensionless parameter :math:`k_z = 2\pi`. This represents only one spatial period of the wave. :math:`TE_{10}`, :math:`TE_{01}` and :math:`TE_{21}`:

Reviewer: ```suggestion
Here are Paraview visuals of different modes. In order to maximise the resolution, we calculated the parameters with :math:`\lambda = L = 1\, \mathrm{m}`, which is equivalent to fix the dimensionless parameter :math:`k_\mathrm{z} = 2\pi`. This represents only one spatial period of the wave. Results of :math:`\mathrm{TE}_{10}`, :math:`\mathrm{TE}_{01}` and :math:`\mathrm{TE}_{21}` are presented below:
```

.. image:: images/TE10.png
    :alt: TE10
    :align: center
    :name: TE10
    :width: 500

.. image:: images/TE01.png
    :alt: TE01
    :align: center
    :name: TE01
    :width: 500

.. image:: images/TE32.png
    :alt: TE32
    :align: center
    :name: TE32

>     :width: 500

Reviewer: These could be made as a table-like composed figure, with the first column for the modes and the second the images.

>     :width: 500

Reviewer: - [ ] The size of the colour-bar and its font could be increased. :)
- [ ] Can you also match all the color schemes to be uniform? In the mesh adaptation section, it is a different colormap.

Order of the FEM solver
~~~~~~~~~~~~~~~~~~~~~~~

> You can also change the order of the polynomes used in the Finite Element Method on line 39 : ``set electromagnetics trial order = x``

Reviewer 3: Add the .prm subsection to be more explicit.

.. note::

>     If you change the trial order, you have to change the test order on line 40 : ``set electromagnetics test order  = y``. in order to have :math:`test \quad order = trial \quad order + 1`

Reviewer: ```suggestion
Degree of the FEM Solver
~~~~~~~~~~~~~~~~~~~~~~~~
You can also change the degree of the polynomes used in the Finite Element Method on line 39: ``set electromagnetics trial degree = 1``

.. caution::
    If you change the trial degree, you have to change the test degree on line 40: ``set electromagnetics test degree  = 2`` in order to have :math:`test \, degree = trial \, degree + 1`.
```

Dielectric caracteristics of the medium

> ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Reviewer: ```suggestion
Dielectric Characteristics of the Medium
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reviewer 3: This section should be named Physical Properties because it is the subsection associated to it in the .prm.

> You can also change the physical properties of the medium on the subsection line 43. These are the default settings for air or void:

Reviewer: ```suggestion
You can also change the ``physical properties`` of the medium at the subsection (line 43). These are the default settings for air or void:
```

.. code-block:: text
    
    subsection physical properties
        set number of fluids = 1
        subsection fluid 0
            set electric conductivity model = constant
            et electric conductivity       = 0.

            set electric permittivity model     = constant
            set electric permittivity real part = 1.
            set electric permittivity imag part = 0.

            set magnetic permeability model     = constant
            set magnetic permeability real part = 1.
            set magnetic permeability imag part = 0.
        end
    end

.. Note::
    Adding an imaginary part to the permeability and the permittivity makes the medium dissipative. 
    
Here is a comparison of the result in a non-dissipative medium (air) and a disspative one with :math:`\Im(\mu_r)=0.06 \ , \ \Im(\varepsilon_r)=0.06`:

.. image:: images/TE10.png
    :alt: TE10V2
    :align: center
    :name: TE10V2
    :width: 500

.. image:: images/Dissipative_medium.png
    :alt: Dissipative medium
    :align: center
    :name: Dissipative medium

>     :width: 500

Reviewer: ```suggestion
.. Note::
    Adding an imaginary part to the permeability and the permittivity makes the medium dissipative. 
    
    Here is a comparison of the result in a non-dissipative medium (air) and a disspative one with :math:`\Im(\mu_\mathrm{r})=0.06 \ , \ \Im(\varepsilon_\mathrm{r})=0.06`:

    .. image:: images/TE10.png
        :alt: TE10V2
        :align: center
        :name: TE10V2
        :width: 500

    .. image:: images/Dissipative_medium.png
        :alt: Dissipative medium
        :align: center
        :name: Dissipative medium
        :width: 500
```

> 

Reviewer: The `Running the Simulation` section is missing. Can add it with a recommended number of cores and the approximate run time? Here is an example of it:
  <img width="855" height="334" alt="Image" src="https://github.com/user-attachments/assets/7453dcf6-6c4a-48dd-8fbf-a29687d49280" />

-----------------------
Results and Discussion
-----------------------

> Here is the error of the default setting simulation (:math:`\lambda = L`) depending on the mesh adaptation and the order of the simulation :

Reviewer: ```suggestion
Here is the error of the default setting simulation (:math:`\lambda = L`) in function of the element size (:math:`h`) and the polynomial degree (:math:`p`):
```

> Here is the error of the default setting simulation (:math:`\lambda = L`) depending on the mesh adaptation and the order of the simulation :

Reviewer 3: The figure you use to show convergence is the one I created for my paper. In it the wavelength is not 1. You can see it from the figure `Mesh_adaptation.png`

.. image:: images/model_validation.png
    :alt: final graph
    :align: center

> As predicted, the error is on the order :math:`h(o^{k+1})` for a simulation on the order k.

Reviewer: I've changed it for $p$ to match with the figure.
```suggestion
As expected, the error is :math:`\mathcal{O}(h^{p+1})`.
```

> As predicted, the error is on the order :math:`h(o^{k+1})` for a simulation on the order k.

Reviewer 3: ```suggestion
As expected, the error is :math:`\mathcal{O}(h^{p+1})` in the :math:`\|L^2\|`  norm for the interior trial space, which validate the implementation of the solver. 
```
